incapsula/resource_incap_rule.go

```diff
@@ -0,0 +1,142 @@
+package incapsula
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceIncapRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceIncapRuleCreate,
+		Read:   resourceIncapRuleRead,
+		Update: resourceIncapRuleUpdate,
+		Delete: resourceIncapRuleDelete,
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				idSlice := strings.Split(d.Id(), "/")
+				if len(idSlice) != 2 || idSlice[0] == "" || idSlice[1] == "" {
+					return nil, fmt.Errorf("unexpected format of ID (%q), expected site_id/rule_id", d.Id())
+				}
+
+				siteID, err := strconv.Atoi(idSlice[0])
+				ruleID := idSlice[1]
+				if err != nil {
+					return nil, err
+				}
+
+				d.Set("site_id", siteID)
+				d.Set("rule_id", ruleID)
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		Schema: map[string]*schema.Schema{
+			// Required Arguments
+			"site_id": {
+				Description: "Numeric identifier of the site to operate on.",
+				Type:        schema.TypeInt,
+				Required:    true,
+			},
+			"enabled": {
+				Description: "todo",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"priority": {
+				Description: "todo",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"name": {
+				Description: "todo",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+
+			// Optional Arguments
+			"action": {
+				Description: "todo",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"filter": {
+				Description: "todo",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"rule_id": {
```

Comment:
rule_id can be removed as its the primary_key

Reply:
removed